src/messages/mod.rs

```diff
@@ -44,17 +45,35 @@ pub enum Message {
     Hop(SignedRoutingMessage),
 }
 
+#[derive(Debug, Eq, PartialEq, Hash, Deserialize)]
+#[allow(clippy::large_enum_variant)]
+pub enum MessagePeek {
+    /// A message sent between two nodes directly
+    Direct(SignedDirectMessage),
+    /// A message sent across the network (in transit)
+    Hop(PeekedSignedRoutingMessageInfo),
+}
+
 #[allow(clippy::large_enum_variant)]
 pub enum MessageWithBytes {
     Hop(HopMessageWithBytes),
     Direct(SignedDirectMessage, Bytes),
 }
 
 impl MessageWithBytes {
-    pub fn from_bytes(bytes: Bytes) -> Result<Self> {
-        match from_network_bytes(&bytes)? {
-            Message::Hop(msg) => Ok(Self::Hop(HopMessageWithBytes::new_from_parts(msg, bytes))),
-            Message::Direct(msg) => Ok(Self::Direct(msg, bytes)),
+    pub fn peek_from_bytes(bytes: Bytes) -> Result<Self> {
+        match peek_from_network_bytes(&bytes)? {
+            MessagePeek::Hop(msg_peek) => Ok(Self::Hop(HopMessageWithBytes::new_from_parts(
+                None, msg_peek, bytes,
+            ))),
+            MessagePeek::Direct(msg) => Ok(Self::Direct(msg, bytes)),
+        }
+    }
+
+    pub fn full_hop_msg_from_bytes(bytes: &Bytes) -> Result<SignedRoutingMessage> {
```

Comment:
Why is this not a normal method that takes `self`? i.e.:
```rust
pub fn full_hop_msg_from_bytes(&self) -> Result<SignedRoutingMessage> {
    match from_network_bytes(&self.bytes) {
        ..
    }
}
```

Reply:
Also it should probably be called `signed_routing_message_from_bytes` for consistency.

Reply:
Then it would need to be a method of `HopMessageWithBytes`.
I wanted to keep this function and `peek_from_bytes` together, maybe not a good idea?

Reply:
It's not very idiomatic. Associated functions usually return `Self` or something that contains `Self`. Also you only call this function from `deserialize_signed_routing_message`, so why not just do the deserialization inline there and not even have this function?